python/sedUtils.py

```diff
@@ -0,0 +1,273 @@
+    rest of these routines. sedDir (if set) can be the root directory for the files in the list."""
+    # Set up the sedlists dictionary to hold the sed names. We'll store them all keyed under 'any'.
+    sedlists = {}
+    sedlists['any'] = deepcopy(inputfileList)
+    # If the root directory is set, add it to the input file names.
+    if sedDir != None:
+        ifiles = []
+        for i in inputfileList:
+            ifiles.append(os.path.join(sedDir, i))
+        inputfileList = ifiles
+    # Read the files.
+    sedDict = {}
+    for filename, s in zip(inputfileList, sedlists['any']):
+        sedDict[s] = Sed()
+        sedDict[s].readSED_flambda(filename)
+    # Return the sed dictionary and the dictionary containing the lists of each type of object.
+    return sedDict, sedlists
+
+
+def makeRedshiftedSeds(sedDict, sedlists, redshifts):
+    """Redshift the quasar, galaxies and SN by the amounts given in redshifts.
+    If redshift is a numpy array, all objects except stars are redshifted to the same values.
+    If redshift is a dictionary containing numpy arrays in the 'quasar', 'sn' or 'galaxies' keys, then
+    those redshifts are applied to those objects only. """
+    # Check what kind of redshift object we received.
+    if isinstance(redshifts, dict):
+        # Loop over all the object types in the redshift dictionary.
+        for objtype in redshifts.keys():
+            # Then loop over all redshifts for this type.
+            newSedsForList = []
+            for z in redshifts[objtype]:
+                # And add new SEDs redshifted to this z for this type of object.
+                for s in sedlists[objtype]:
+                    # Make the new name for the new SED at this redshift.
+                    newsedname = s + '_Z_%.3f' %(z)
+                    # Add it to the SED dictionary.
+                    sedDict[newsedname] = redshiftSingleSED(sedDict[s], z)
+                    newSedsForList.append(newsedname)
+            # Done redshifting all objects of this type; updated sedlist.
+            sedlists[objtype] += newSedsForList
+    else:
+        # Using the same redshift list for everything (except stars).
+        for z in redshifts:
+            for objtype in sedlists.keys():
+                newSedsForList = []
+                if objtype == 'stars':
+                    # Skip stars.
+                    continue
+                for s in sedlists[objtype]:
+                    newsedname = s + '_Z_%.3f' %(z)
+                    sedDicts[newsedname] = redshiftSingleSED(sedDict[s], z)
+                    newSedsForList.append(newsedname)
+                sedlists[objtype] += newSedsForList
+    return sedDict, sedlists
+
+
+def redshiftSingleSED(sed_in, z):
+    # Make a copy of the input SED, as otherwise we will overwrite original arrays.
+    sed_out = deepcopy(sed_in)
+    sed_out.redshiftSED(z, dimming=False)
+    return sed_out
+
+
+def matchSedsBp(sedDict, bpDict, refFilter=None):
```

Comment:
There is now a class in sims_photUtils called BandpassDict which reads in a bunch of bandpasses, and then automatically resamples them to the same wavelength grid and calculates their phi array.  It uses Sed.manyMagCalc to calculate the magnitude of any Sed in all of the bandpasses simultaneously.  Maybe that is worth using here?